elasticsearch/configure.sls

```diff
@@ -0,0 +1,109 @@
+{% from "elasticsearch/map.jinja" import elasticsearch with context %}
+
+include:
+  - .service
+  - .repository
+
+# Create folder and set permissions
+{% set conf_dir = elasticsearch.conf_folder %}
+{% set log_dir = elasticsearch.log_folder %}
+
+{% for dir in (conf_dir, log_dir) %}
+{% if dir %}
+{{ dir }}:
+  file.directory:
+    - user: elasticsearch
+    - group: elasticsearch
+    - mode: 0700
+    - makedirs: True
+{% endif %}
+{% endfor %}
+
+# Update the I/O scheduler if using SSD to improve write throughput https://www.elastic.co/guide/en/elasticsearch/guide/current/hardware.html
+{% for device_name in salt.grains.get('SSDs') %}
+update_io_scheduler_for_{{device_name}}:
+  cmd.run:
+    - name: echo noop | tee /sys/block/{{ device_name }}/queue/scheduler
+{% endfor %}
+
+# Update Heap Size according to available RAM https://www.elastic.co/guide/en/elasticsearch/reference/current/heap-size.html
+{% set heap_max = salt.grains.get('mem_total', 0) // 2 %}
+{% set heap_size = heap_max if heap_max < 31744 else 31744 %}
+{% if elasticsearch.version == '5.x' %}
+config_jvm_options:
+  file.replace:
```

Comment:
file.comment appears to have been deprecated since 0.17.0 and file.replace is the recommendation.